moonscript/parse.moon

```diff
@@ -32,297 +31,307 @@ Num = Space * (Num / (v) -> {"number", v})
 } = require "moonscript.parse.util"
 
 
-build_grammar = wrap_env debug_grammar, (root) ->
-  _indent = Stack 0
-  _do_stack = Stack 0
+build_grammar_factory = (litmoon=false) ->
```

Comment:
I'm not sure why there's a build grammar factory when build grammar is already effectively a factory. If we just update the existing function we don't have to blast the whole blame of the file by reindenting it as well.